grails-app/services/org/transmartproject/db/RestExportService.groovy

```diff
@@ -27,47 +32,39 @@ class RestExportService {
-        //Maybe make this a marshaller.
-        def returnDataTypeList = []
-        def dataTypesList = []
-        datatypes.each { dataTypeMap ->
-            dataTypeMap.each {key, value->
-                if (key in dataTypesList){
-                    returnDataTypeList.each { map ->
-                        if (key in map.values()){
-                            value.each{it.remove('datatypeCode')}
-                            map.get('cohorts').add([concepts:value])
-                        }
-                    }
-                } else{
-                    dataTypesList.add(key)
-                    def datatypeCode = value.collect({it ->
-                        it.get('datatypeCode')})
-                    value.each{it.remove('datatypeCode')}
-                    def datatypeMap = [dataType:key,
-                                       dataTypeCode: datatypeCode[0],
-                                       cohorts:[[concepts:value]]]
-                    returnDataTypeList.add(datatypeMap)
-                }
+    public List<DataTypeRetrieved> retrieveDataTypes(params) {
+        if (!(params.containsKey('concepts'))) {
+            throw new NoSuchResourceException("No parameter named concepts was given.")
+        }
+
+        if (params.get('concepts') == "") {
+            throw new InvalidArgumentsException("Parameter concepts has no value.")
+        }
+
+        def jsonSlurper = new JsonSlurper()
+        def conceptParameters = params.get('concepts').decodeURL()
+        List<DataTypeRetrieved> dataTypes = []
+        try {
+            def conceptArguments = jsonSlurper.parseText(conceptParameters)
+            int cohortNumber = 0
+            conceptArguments.cohorts.collect { it ->
+                cohortNumber += 1
+                getDataTypes(it, dataTypes, cohortNumber)
             }
+            dataTypes
+        } catch (JsonException e) {
+            throw new InvalidArgumentsException("Given parameter was non valid JSON.")
         }
-        returnDataTypeList
     }
 
-
-    def getDataTypes(List conceptKeysList){
-        List cohortDataTypes = []
-        Map datatypesMap = [:]
-        conceptKeysList.each { conceptKey ->
-            OntologyTerm concept = conceptsResourceService.getByKey(conceptKey)
-            datatypesMap = getHighDimDataType(concept, datatypesMap)
+    private List<DataTypeRetrieved> getDataTypes(Map conceptKeysList, List dataTypes, int cohortNumber) {
+        conceptKeysList.conceptKeys.collect { conceptKey ->
+            getDataType(conceptKey, dataTypes, cohortNumber)
         }
-        cohortDataTypes += datatypesMap
-        cohortDataTypes
     }
 
-    def getHighDimDataType(OntologyTerm term, Map datatypesMap) {
+    private void getDataType(String conceptKey, List dataTypes, int cohortNumber) {
+        OntologyTerm term = conceptsResourceService.getByKey(conceptKey)
         // Retrieve all descendant terms that have the HIGH_DIMENSIONAL attribute
```

Comment:
this comment is intended as a description of this method right?
Then you should put it on top of the method declaration like this:
 
    /**
     * description of the functionality
     */
     method() {}